### PR TITLE
Friendshipv1

### PR DESCRIPTION
**Pull request by yahir91**
In this milestone, we have to be able to make friends.

- A model friendship with the corresponding associations was added.
- The option to add a friend was added to the views.
- A user can accept and reject a friend request.
- A new page was called friendship was created where you can see your friend request.

### CODE REVIEW

File: app/controllers/friendships_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    user = User.find(params[:user_id])
+    current_user.confirm_friend(user)
+    redirect_to friendships_path, notice: 'You confirmed a friend request.'
+  end
+
+  def reject
+    user = User.find(params[:user_id])
+    current_user.destroy_friend(user)
+    redirect_to friendships_path, notice: 'You reject a friend request.'
+  end
```
Comment (kobiyoyo) [inline app/controllers/friendships_controller.rb:27]: Everything connected with data processing should be encapsulated in models and used in controllers.

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, everything works fine, but there are a few more things required. 
### General

- [x] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] Lets change that `Already friends` to `unfriend`

- [x] Let's write unit and integration test for all components using Rspec and capybara
- [x] Check comment below
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## Impressive progress so far 💪! Thanks for making all the changes 👏.

### However, we need to work on some changes before we can merge.

- Please fix all linter errors BEFORE requesting a review.
- Organize your tests in different folders - `spec/controllers`, `spec/models`, etc.
- Write tests for all your models. Don't forget to test associations and validations. _HINT: Use shoulda_matchers gem._
- Please DO NOT commit commented codes. You have commented codes in ` spec/auth_spec.rb` and `spec/friend_spec.rb`.

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Review by stratospherique — APPROVED** [review]
### Great job on implementing the suggested changes :clap: Congrats, the milestone is approved :heavy_check_mark: You may merge :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**

**Pull request merged**